cli/js/web/dom_types.d.ts

```diff
@@ -23,37 +23,30 @@ export type HeadersInit =
   | Headers
```

Comment:
This file contains a ton of overlap with cli/js/lib.deno.shared_globals.d.ts is there some way this code can be shared?

Reply:
Let me think about it.  It is a valid point, and it is slowly killing us by a thousand papercuts.

Reply:
Ok, I found a way... For things that are part of this PR (Event, EventTarget, CustomEvent, and DOMException) I have eliminated them from dom_types and instead am just using lib.deno.shared_globals.d.ts.  I would recommending doing the rest as a subsequent PR.  It does take a little bit of effort to make sure you get it right.  Take a look at the last commit in isolation to get an idea of what I had to do.